rts/Sim/Units/UnitTypes/Builder.cpp

```diff
@@ -731,11 +748,12 @@ bool CBuilder::StartBuild(BuildInfo& buildInfo, CFeature*& feature, bool& inWait
 
 				if (cu == nullptr)
 					continue;
+				if (!teamHandler.Ally(allyteam, cu->allyteam)
```

Comment:
I have doubts. Here's the callouts to put in some gadget that ceasefires both teams for a test (the rest of the test setup stays the same as previous since ZK doesn't have allied victory AFAIK and LoS isn't shared):
```lua
Spring.SetAlly(0, 1)
Spring.SetAlly(1, 0)
```

Reply:
doubts about what?

Reply:
Doubts that checking whether the teams are allied is correct to do here.

Reply:
Anyway, the suggested change seems to be missing a parenthesis.